src/app/reporting/EventLoggingTypes.h

```diff
@@ -0,0 +1,273 @@
+    kTag_Message = 2  ///< A string containing the actual debug message.
+};
+
+// Structures used to describe in detail additional options for event encoding
+
+/**
+ * @brief
+ *   The structure that provides a full resolution of the cluster.
+ */
+struct DetailedRootSection
+{
+    /**
+     * Default constructor
+     */
+    DetailedRootSection(void){};
+    chip::NodeId mNodeId;
+    chip::EndpointId mEndpointId;
+};
+
+/**
+ * @brief
+ *   The validity and type of timestamp included in EventOptions.
+ */
+typedef enum TimestampType
+{
+    kTimestampType_Invalid = 0,
+    kTimestampType_System,
+    kTimestampType_UTC
+} TimestampType;
+
+/**
+ * @brief
+ *   The union that provides an application set system or UTC timestamp.
+ */
+union Timestamp
+{
+    /**
+     * Default constructor.
+     */
+    Timestamp(void) : systemTimestamp(0){};
+
+    /**
+     * UTC timestamp constructor.
+     */
+    Timestamp(utc_timestamp_t aUtc) : utcTimestamp(aUtc){};
+
+    /**
+     * System timestamp constructor.
+     */
+    Timestamp(timestamp_t aSystem) : systemTimestamp(aSystem){};
+
+    timestamp_t systemTimestamp;  ///< System timestamp.
+    utc_timestamp_t utcTimestamp; ///< UTC timestamp.
+};
+
+/**
+ *   The structure that provides options for the different event fields.
+ */
+class EventOptions
+{
+public:
+    EventOptions(void);
+    EventOptions(bool);
+    EventOptions(timestamp_t);
+    EventOptions(utc_timestamp_t);
+    EventOptions(timestamp_t, bool);
+    EventOptions(utc_timestamp_t, bool);
+
+    Timestamp timestamp; /**< A union holding either system or UTC timestamp. */
+
+    DetailedRootSection * eventSource; /**< A pointer to the detailed resolution of the cluster instance.  When NULL, the event
+                                          source is assumed to come from the resource equal to the local node ID, and from the
+                                          default instance of the cluster. */
+
+    PriorityLevel relatedPriority; /**< EventPriority of the Related Event Number.  When this event and the related event are of the
+                                           same priority, the field may be omitted.  A value of kPriorityLevel_Invalid implies the
+                                           absence of any related event. */
+
+    TimestampType timestampType; /**< An enum indicating if the timestamp is valid and its type. */
+
+    bool urgent; /**< A flag denoting that the event is time sensitive.  When set, it causes the event log to be flushed. */
+};
+
+/**
+ * @brief
+ *   Structure for copying event lists on output.
+ */
+
+struct EventLoadOutContext
+{
+    EventLoadOutContext(chip::TLV::TLVWriter & aWriter, PriorityLevel aPriority, uint32_t aStartingEventID);
+
+    chip::TLV::TLVWriter & mWriter;
+    PriorityLevel mPriority;
+    uint32_t mStartingEventNumber;
+    uint32_t mCurrentTime;
```

Comment:
should these be some of the time_t defined above?

Reply:
yep, done, thanks